### PR TITLE
API-12186 add encounter appointment reference

### PR DESCRIPTION
**Pull request by jblefkove-va**

### CODE REVIEW

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/encounter/DatamartEncounter.java [hunk 1]
```diff
@@ -41,6 +41,8 @@
 
   private List<Participant> participants;
```
Comment (taylorrassmannlibertyits) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/encounter/DatamartEncounter.java:43]: Lazy initializer?

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/encounter/R4EncounterController.java [hunk 2]
```diff
@@ -121,7 +121,10 @@ String readRaw(@PathVariable("publicId") String publicId, HttpServletResponse re
         .replaceReferences(
             resource ->
                 Streams.concat(
```
Comment (jhulbert-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/encounter/R4EncounterController.java:123]: Should we be filtering null references from the concated stream?

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/encounter/DatamartEncounter.java (hunk 1 above)
Comment (jhulbert-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/encounter/DatamartEncounter.java:43]: +1

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/encounter/R4EncounterController.java (hunk 2 above)
Comment (jblefkove-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/encounter/R4EncounterController.java:123]: `WitnessProtection` filters out the null references

**Review by taylorrassmannlibertyits — APPROVED** [review]
:shipit:

**Review by jhulbert-va — APPROVED** [review]

**Pull request merged**